### ISSUE TITLE
Unable to use DbFunction in 3.0

### ISSUE THREAD

**Issue by cwoolum** [aspnet/EntityFramework.Docs#1747]
I am trying to use `DatePart` in EfCore 3.0 but it seems to fail due to the `dbo` schema being prepended to the function name.

### Steps to reproduce

``` C#
public class AnalyticsDbContext : DbContext
{
    public AnalyticsDbContext(DbContextOptions options) : base(options)
    {
    }

    public DbSet<AggregatedMetric> AggregatedMetrics { get; set; }

    [DbFunction("DatePart", Schema = "")]
    public static int? DatePart(string datePartArg, DateTime? date) => throw new Exception();

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);   
    }
}
```

I always get the error `{"Cannot find either column \"dbo\" or the user-defined function or aggregate \"dbo.DatePart\", or the name is ambiguous."}` even though my definition shows an empty schema.

I would expect that the `DbFunctionAttribute` would be respected and the schema would be blank. I halve also tried to use the fluent API to register with the same problem

```C#
modelBuilder
    .HasDbFunction(typeof(DateExtensions).GetMethod(nameof(DateExtensions.DatePart)))
    .HasName("DatePart")
    .HasSchema("");
```

### Further technical details

EF Core version: 3.0.0-rc1
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Target framework: (e.g. .NET Core 3.0)
Operating system:
IDE: (e.g. Visual Studio 2019 16.3)

**Comment by smitpatel** [comment]
@cwoolum - In EF Core 3.0, we removed assumption that empty schema means built-in function. So null or empty string as schema means it's default schema which would be dbo. in order to configure to use built in function it requires you to map custom translation like follows.
```C#
            modelBuilder
                .HasDbFunction(typeof(MyContext).GetMethod(nameof(MyContext.DatePart)))
                .HasTranslation(args => SqlFunctionExpression.Create("DatePart", args, typeof(int?), null));
```
Note that above would also fail because datepartarg will be printed as string which is not valid in SqlServer. The custom translation also need to take care of it but currently blocked by https://github.com/aspnet/EntityFrameworkCore/issues/17918

Overall for this scenario see https://github.com/aspnet/EntityFrameworkCore/issues/17268 for various ways we plan to improve this experience.

**Comment by smitpatel** [comment]
I transferred this issue here to document the way to map to a built-in function which is different from #504

**Comment by smitpatel** [comment]
Also document the breaking change that null or empty schema means default schema defined in model.

**Comment by AndriySvyryd** [comment]
@smitpatel The breaking change note should be added ASAP

**Comment by ajcvickers** [comment]
@smitpatel Duplicate of #500?
